Java/Dynamic-Programming/DP.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Created Date:  June 1st 2019
+ * @author ganesh chowdhary sadanala
+ * Email-ID: ganesh8374387106@gmail.com
+ * -----
+ * Series on Dynamic programming
+ */
+
+//Fibanocci number using recursion
+//Algorithm:
+/*  Fib(n)
+    if n<=1
+       return n
+    return Fib(n-2)+Fib(n-1) 
+
+  Same value is calculated number of times incurring cpu cycles and time cost as well as stack resource.
+Two ways we can eliminate this:
+1.Memoization/TopDown - Look in the data structure(a look up table in the memory ),
+if present
+  return that value
+else
+  find and place in the look up table
+
+2nd method will be dealt after some hands on coding
+*/
```

Comment:
Damn nice job on the comments and sticking with Camel Case! 👍

Reply:
Thanks buddy! there are a lot to come.

Reply:
Sure! I will comment the links to amazing competitive problems.